nav2_map_server/src/occ_grid_loader.cpp

```diff
@@ -152,25 +160,66 @@ nav2_util::CallbackReturn OccGridLoader::on_configure(const rclcpp_lifecycle::St
     RCLCPP_ERROR(
       node_->get_logger(), "Failed to load image file %s for reason: %s",
       loadParameters.image_file_name.c_str(), e.what());
+    response->result = nav2_msgs::srv::LoadMap::Response::RESULT_INVALID_MAP_DATA;
+    return false;
+  }
+  return true;
+}
+
+nav2_util::CallbackReturn OccGridLoader::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(node_->get_logger(), "OccGridLoader: Configuring");
+
+  // initialize Occupancy Grid msg - needed by loadMapFromYaml
+  msg_ = std::make_unique<nav_msgs::msg::OccupancyGrid>();
 
-    throw std::runtime_error("Failed to load map image file.");
+  if (!loadMapFromYaml(yaml_filename_)) {
+    throw std::runtime_error("Failed to load map yaml file: " + yaml_filename_);
   }
 
-  // Create a service callback handle
+  // Create GetMap service callback handle
   auto handle_occ_callback = [this](
     const std::shared_ptr<rmw_request_id_t>/*request_header*/,
     const std::shared_ptr<nav_msgs::srv::GetMap::Request>/*request*/,
     std::shared_ptr<nav_msgs::srv::GetMap::Response> response) -> void {
-      RCLCPP_INFO(node_->get_logger(), "OccGridLoader: Handling map request");
+      RCLCPP_INFO(node_->get_logger(), "OccGridLoader: Handling GetMap request");
       response->map = *msg_;
     };
 
   // Create a service that provides the occupancy grid
   occ_service_ = node_->create_service<nav_msgs::srv::GetMap>(service_name_, handle_occ_callback);
 
+  // Create the load_map service callback handle
+  auto load_map_callback = [this](
```

Comment:
I'm prefer if this was a member function rather than a lambda, but up to you

Reply:
I tend to agree, but the "GetMap" service call was also defined as a lambda so I was being consistent with the existing code. If I change this, I would want to change both. I don't think it's necessary though.

Reply:
Just preference, I'd just do both but I'm not blocking over it. I'm generally anti-lambda when its just operating with class members or setting flags, it seems like overuse to me.